Patchin: in-house
Patch status: Solaris-specific; not suitable for upstream

Need to add sample pem files to make component tests work.

--- libpki-0.8.9/src/token.c	Tue Nov 10 13:50:28 2015
+++ libpki-0.8.9/src/token.c	Wed Jan 20 14:21:10 2016
@@ -120,7 +120,7 @@
 	if (PKI_get_init_status() == PKI_STATUS_NOT_INIT) PKI_init_all();
 
 	// Sets the default callback for getting the credentials
-	PKI_TOKEN_cred_set_cb(tk, PKI_TOKEN_cred_cb_stdin, NULL);
+	PKI_TOKEN_cred_set_cb(tk, PKI_TOKEN_cred_cb_env, NULL);
 
 	// Initializes the token
 	PKI_TOKEN_init( tk, NULL, NULL );
--- libpki-0.8.9/etc/token.d/default.xml	Tue Nov 10 13:50:29 2015
+++ libpki-0.8.9/etc/token.d/default.xml	Wed Jan 20 14:31:35 2016
@@ -12,10 +12,10 @@
   <!-- CA Certificate -->
   <pki:cacert>file:///$HOME/.libpki/certs/cacert.pem</pki:cacert>
   <!-- Certificates -->
-  <pki:othercerts>file::///$HOME/.libpki/certs/other-certs.pem</pki:othercerts>
-  <pki:trustedcerts>file::///$HOME/.libpki/certs/trusted-certs.pem</pki:trustedcerts>
+  <!-- <pki:othercerts>file::///$HOME/.libpki/certs/other-certs.pem</pki:othercerts> -->
+  <!-- <pki:trustedcerts>file::///$HOME/.libpki/certs/trusted-certs.pem</pki:trustedcerts> -->
   <!-- Where the Password should be read from -->
-  <pki:passin>stdin</pki:passin>
+  <pki:passin>env</pki:passin>
   <!-- ... or simply specify the password here -->
   <!-- <pki:password></pki:password> -->
 </pki:tokenConfig>
--- libpki-0.8.9/etc/.libpki/certs/cacert.pem	1969-12-31 16:00:00.000000000 -0800
+++ libpki-0.8.9/etc/.libpki/certs/cacert.pem	2016-01-15 15:57:07.721997458 -0800
@@ -0,0 +1,25 @@
+-----BEGIN CERTIFICATE-----
+MIIEOTCCAyGgAwIBAgIJAMvfuAXWxNmsMA0GCSqGSIb3DQEBCwUAMIGyMQswCQYD
+VQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTEUMBIGA1UEBwwLU2FudGEgQ2xh
+cmExHTAbBgNVBAoMFE9yYWNsZSBBbWVyaWNhLCBJbmMuMRQwEgYDVQQLDAtVc2Vy
+bGFuZCBDQTEfMB0GA1UEAwwWdXNlcmxhbmQudXMub3JhY2xlLmNvbTEiMCAGCSqG
+SIb3DQEJARYTdXNlcmxhbmRAb3JhY2xlLmNvbTAeFw0xNjAxMTEyMTM4MzdaFw0x
+NjAyMTAyMTM4MzdaMIGyMQswCQYDVQQGEwJVUzETMBEGA1UECAwKQ2FsaWZvcm5p
+YTEUMBIGA1UEBwwLU2FudGEgQ2xhcmExHTAbBgNVBAoMFE9yYWNsZSBBbWVyaWNh
+LCBJbmMuMRQwEgYDVQQLDAtVc2VybGFuZCBDQTEfMB0GA1UEAwwWdXNlcmxhbmQu
+dXMub3JhY2xlLmNvbTEiMCAGCSqGSIb3DQEJARYTdXNlcmxhbmRAb3JhY2xlLmNv
+bTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAMzwfyDlWbVLMKqmAUhG
+KuRhjLS03DrOumIuYwI1oQvLXZz4u4wnbMc5GgViUNw04AC7geJjWYunI1U5LAG5
+dQpbrMK1xL1iDMA0vc8EHvuUFY59tzjcGY1CAYtCNok1dau8FUWBfxV9wWEW00t3
+PEGREPuz5UtUcor7KX+JJ6r9/sT7pAKj6Y0G4v77RBDLtJ6hBXEJhqZ8eahtzOv3
+DBDSlOEimkCLxvlZnk0vTgpKCRmksIp7xq4MM61JiWuKNx9oJUgrEMgk2gP8Ox09
+RX1sG3GvDwW3L8rxKxCftWXPo22GV9OB5FAiHoqi+aLaSMXxHEyAlwXX2Fs9ZV8U
+iKkCAwEAAaNQME4wHQYDVR0OBBYEFJizE+yWESOeJaEhBGEkYSlrMldCMB8GA1Ud
+IwQYMBaAFJizE+yWESOeJaEhBGEkYSlrMldCMAwGA1UdEwQFMAMBAf8wDQYJKoZI
+hvcNAQELBQADggEBAI5UFjJJtPP4tsEdzAiH8dvEBG2tuDkPRpxbVd1G8lfDFLsD
+9MgawSeM8F4e0NQg4mP7/sshp3JpZ+NJMMNZkY57UYs28FuXa7WKx05aiMaXCxOH
+IDs/H8YCfcZq7JcpX5mdiiRqCU0w3h0/1w0b38JGSQF5tj6InjkNqD+jqrYiQeTj
+MPysxoAGiEDIq0AOuYgAG0kD7WytQRj6p3uxR2eWGIE2GYr715DiBjleaZoX1YQM
+AXTPEyqtES4ma+J3mRFUNWAvQTfsBpYS4/a32H2j+5tZchJd2T3AUvNcG5p+PvZB
+Ckks3qIQUXinjtZLtO3ZpSySW3XjRUeWpW+ij+A=
+-----END CERTIFICATE-----
--- libpki-0.8.9/etc/.libpki/certs/cert.pem	1969-12-31 16:00:00.000000000 -0800
+++ libpki-0.8.9/etc/.libpki/certs/cert.pem	2016-01-15 15:57:07.747595458 -0800
@@ -0,0 +1,72 @@
+Certificate:
+    Data:
+        Version: 1 (0x0)
+        Serial Number: 0 (0x0)
+    Signature Algorithm: sha1WithRSAEncryption
+        Issuer: C=US, ST=California, L=Santa Clara, O=Oracle America, Inc., OU=Userland CA, CN=userland.us.oracle.com/emailAddress=userland@oracle.com
+        Validity
+            Not Before: Jan 11 21:40:47 2016 GMT
+            Not After : Jan 10 21:40:47 2021 GMT
+        Subject: C=US, ST=California, L=Santa Clara, O=Oracle America, Inc., OU=Userland, CN=userland.us.oracle.com/emailAddress=userland@oracle.com
+        Subject Public Key Info:
+            Public Key Algorithm: rsaEncryption
+                Public-Key: (2048 bit)
+                Modulus:
+                    00:f1:dc:ab:d3:72:2d:8f:7b:71:7d:15:46:32:8a:
+                    da:97:db:91:70:ce:37:a8:18:dd:1f:aa:5d:83:c0:
+                    bc:44:23:fb:b0:da:2c:14:e8:b0:56:05:ef:98:22:
+                    54:28:a0:53:c4:24:7a:1c:d3:23:97:8d:8f:ea:ee:
+                    ff:6b:d4:b8:65:0f:33:28:35:3c:71:9c:6c:e5:bf:
+                    03:87:5c:51:5e:5a:23:da:24:8e:6d:92:56:c7:93:
+                    d5:73:54:42:aa:59:23:8b:ce:87:ea:bb:5a:06:c4:
+                    ac:ce:6a:44:ba:b7:14:6b:f6:88:49:8d:3f:d5:b7:
+                    ab:4f:2a:8a:32:15:4a:87:78:e6:d4:18:49:b0:6c:
+                    c3:29:25:ab:15:82:b1:e0:59:ef:d7:ef:c2:5e:e3:
+                    b7:bb:63:c3:d4:f1:c8:fe:8e:eb:fd:5c:6c:63:7d:
+                    17:88:62:24:86:5d:cb:ad:93:e2:fb:4d:02:c3:15:
+                    49:43:d0:15:5d:21:63:9b:d1:49:41:f6:b5:52:c9:
+                    5e:3b:42:61:53:dd:20:6c:9e:4b:4e:ab:c2:e1:15:
+                    30:12:ad:73:f1:42:ad:4e:84:10:ec:29:4d:55:69:
+                    ee:32:88:8b:81:50:09:d5:dc:9b:90:87:cb:42:8d:
+                    5b:1f:94:e7:10:be:81:1a:13:2a:92:df:8d:2d:78:
+                    5e:89
+                Exponent: 65537 (0x10001)
+    Signature Algorithm: sha1WithRSAEncryption
+         91:5d:fb:3e:f8:4d:45:98:18:68:d1:12:a0:34:93:30:ce:80:
+         fa:13:43:bb:65:90:07:6c:e1:c1:7b:b8:30:72:81:02:e9:a1:
+         ae:1f:3a:f4:72:20:bc:9a:90:84:5e:0e:1f:db:15:aa:90:cc:
+         0a:ad:7b:08:0e:88:99:3d:93:31:f5:8f:d2:3f:bf:12:2a:45:
+         96:ad:48:e6:2f:37:93:1d:ae:95:61:a2:01:ec:5d:ae:43:b4:
+         7e:b7:02:92:6e:98:0f:94:9c:68:e5:f8:b4:ea:89:5a:55:fb:
+         46:81:0c:1d:2a:74:95:95:ba:ad:6a:a4:08:40:2e:8d:e1:39:
+         4c:74:c4:3c:39:68:18:93:da:2e:f1:ea:6e:96:8c:54:9f:33:
+         58:76:2a:d9:a9:25:fc:87:03:f2:d8:e1:91:17:fe:55:8e:9a:
+         c0:dd:d2:98:02:ae:65:b3:79:a6:98:f3:01:4c:d7:d1:f1:00:
+         17:20:5c:04:48:77:5d:84:c0:bc:99:7a:4e:7a:0f:31:1b:b9:
+         be:c4:4b:5b:b1:e5:3d:8b:34:f1:5a:c0:97:c7:b6:be:70:dc:
+         74:03:fe:be:96:2d:be:0c:0e:0b:5b:67:d4:1d:21:cf:21:1d:
+         67:b5:b6:4a:50:1e:bd:d1:f6:00:87:c3:cb:98:9f:6c:3a:d4:
+         f3:a2:ee:d1
+-----BEGIN CERTIFICATE-----
+MIID1zCCAr8CAQAwDQYJKoZIhvcNAQEFBQAwgbIxCzAJBgNVBAYTAlVTMRMwEQYD
+VQQIDApDYWxpZm9ybmlhMRQwEgYDVQQHDAtTYW50YSBDbGFyYTEdMBsGA1UECgwU
+T3JhY2xlIEFtZXJpY2EsIEluYy4xFDASBgNVBAsMC1VzZXJsYW5kIENBMR8wHQYD
+VQQDDBZ1c2VybGFuZC51cy5vcmFjbGUuY29tMSIwIAYJKoZIhvcNAQkBFhN1c2Vy
+bGFuZEBvcmFjbGUuY29tMB4XDTE2MDExMTIxNDA0N1oXDTIxMDExMDIxNDA0N1ow
+ga8xCzAJBgNVBAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRQwEgYDVQQHDAtT
+YW50YSBDbGFyYTEdMBsGA1UECgwUT3JhY2xlIEFtZXJpY2EsIEluYy4xETAPBgNV
+BAsMCFVzZXJsYW5kMR8wHQYDVQQDDBZ1c2VybGFuZC51cy5vcmFjbGUuY29tMSIw
+IAYJKoZIhvcNAQkBFhN1c2VybGFuZEBvcmFjbGUuY29tMIIBIjANBgkqhkiG9w0B
+AQEFAAOCAQ8AMIIBCgKCAQEA8dyr03Itj3txfRVGMoral9uRcM43qBjdH6pdg8C8
+RCP7sNosFOiwVgXvmCJUKKBTxCR6HNMjl42P6u7/a9S4ZQ8zKDU8cZxs5b8Dh1xR
+Xloj2iSObZJWx5PVc1RCqlkji86H6rtaBsSszmpEurcUa/aISY0/1berTyqKMhVK
+h3jm1BhJsGzDKSWrFYKx4Fnv1+/CXuO3u2PD1PHI/o7r/VxsY30XiGIkhl3LrZPi
++00CwxVJQ9AVXSFjm9FJQfa1UsleO0JhU90gbJ5LTqvC4RUwEq1z8UKtToQQ7ClN
+VWnuMoiLgVAJ1dybkIfLQo1bH5TnEL6BGhMqkt+NLXheiQIDAQABMA0GCSqGSIb3
+DQEBBQUAA4IBAQCRXfs++E1FmBho0RKgNJMwzoD6E0O7ZZAHbOHBe7gwcoEC6aGu
+Hzr0ciC8mpCEXg4f2xWqkMwKrXsIDoiZPZMx9Y/SP78SKkWWrUjmLzeTHa6VYaIB
+7F2uQ7R+twKSbpgPlJxo5fi06olaVftGgQwdKnSVlbqtaqQIQC6N4TlMdMQ8OWgY
+k9ou8epuloxUnzNYdirZqSX8hwPy2OGRF/5VjprA3dKYAq5ls3mmmPMBTNfR8QAX
+IFwESHddhMC8mXpOeg8xG7m+xEtbseU9izTxWsCXx7a+cNx0A/6+li2+DA4LW2fU
+HSHPIR1ntbZKUB690fYAh8PLmJ9sOtTzou7R
+-----END CERTIFICATE-----
--- libpki-0.8.9/etc/.libpki/private/key.pem	1969-12-31 16:00:00.000000000 -0800
+++ libpki-0.8.9/etc/.libpki/private/key.pem	2016-01-15 15:57:07.794138570 -0800
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
+MIIEpgIBAAKCAQEA8dyr03Itj3txfRVGMoral9uRcM43qBjdH6pdg8C8RCP7sNos
+FOiwVgXvmCJUKKBTxCR6HNMjl42P6u7/a9S4ZQ8zKDU8cZxs5b8Dh1xRXloj2iSO
+bZJWx5PVc1RCqlkji86H6rtaBsSszmpEurcUa/aISY0/1berTyqKMhVKh3jm1BhJ
+sGzDKSWrFYKx4Fnv1+/CXuO3u2PD1PHI/o7r/VxsY30XiGIkhl3LrZPi+00CwxVJ
+Q9AVXSFjm9FJQfa1UsleO0JhU90gbJ5LTqvC4RUwEq1z8UKtToQQ7ClNVWnuMoiL
+gVAJ1dybkIfLQo1bH5TnEL6BGhMqkt+NLXheiQIDAQABAoIBAQCjCZBCfws7V4lP
+OlIRrAxipb5K/pgCFpCo0d0r/XZrsJ3vv4JLrfeAZYv3T9zPbhRJehVe7O1nMiQ7
+yFQhyklqxm5Dwm82Xq4ByVuFmbO4GKdmW5IvRqB89KjEj55I2sHdcqGomBa9zUJ3
++qbN/rAjfxKsEi1f0eoG+bQiwrt0ybmqshFeW3c9c6HVZBT9nC1v/+Z6Wz0Na0E+
+eixIrZww3sxz/fsc5iUGehCB7z+poVvdHLVgsZ9fh3UEZfua9AqlHIfQujx/rtVw
+flk99rWjjYWozFN65dyXhSKYDIXhMg/61L6iFUdsu1pYNt9lf6FrBKDwhcPGLWDe
+sN20G2oBAoGBAP9JUnX9JKEkRkY0cTVNynMBfFH69kmG5ZNZuJmqZNMLLH2m+C7P
+GptHH5WbY+glDFPEqM3tlxUnB/yUrCMRm9EsCf3GKoEL516FZkVQ1dyFKpcYkNv2
+biYfhAhWG+DxV2jAC1qkcr69fE2fIUrRnBo3sJPSDlsBsOaU/phEDuwhAoGBAPKJ
+vip62Qbh+7YXoc7k9HXw4PfqjXAx01hT2IBDxwx5FUHrzT6D+9WGgBpxFV9quyz6
+936npQC270WDEMxLo4NiVXd9b/ji0o2+K43+BXEzEX8VkTeMWfveu6Szfq0bskaC
+3FrtgebMwBXUR9Od4WTRX45s+O3UI2jnX52mIOVpAoGBAJpP+P+HUF/zrGL9ijz9
+U9Xm3P41JDDX0OgcAOEqRZ3okDE3/6dT2zJncKkfb+nAk04Df0JTTtMjb8l3GQJt
+v3FWHIgBBGpTBr11B5f4v/13pYIGAHJl6Lk0rbvYSSvU1tBj3bkh1JaFtaC6AwbV
+43VZmmwQOks6n3MSNz9VTGvhAoGBAMFF9M2jQckTUmPEJ0RY/VnzD0NgnT9X5URQ
+5Ikp+be90avLfa4gvRFl8jWgAdkck3X/tDQw4fNU1A1/qWYzrGSpetQSmKUsE3+y
+mMC5OZsAiCPOiv9KD/b/gmZ4Ry0pkgJQi8rF06uZZETA+k/vU0rk8hHsUt4rG+Cf
+6HkXhsVpAoGBAOYLL5vjqV1DyX4smWk4SHnZS27E0fT8yCCMh0ZGRv9AZNTixs9f
+jZ8YI74uVd6WzrR720H/VLCqSQk7hqzZXGF589xlDENQoMxrdNy7oDYblVjv8/Rk
+T5YG6xxJnnCfM30waRWG3fatt6+c+Hy2SfVDpiJnP2+pbMzxgqnznhIq
+-----END RSA PRIVATE KEY-----
